Perlang.Interpreter/PerlangInterpreter.cs

```diff
@@ -906,14 +908,32 @@ public VoidObject VisitWhileStmt(Stmt.While stmt)
                 case STAR:
                     CheckNumberOperands(expr.Operator, left, right);
                     return leftNumber * rightNumber;
+                case STAR_STAR:
+                    CheckNumberOperands(expr.Operator, left, right);
+
+                    if (leftNumber is float || leftNumber is double || rightNumber is float || rightNumber is double)
+                    {
+                        return Math.Pow(leftNumber, rightNumber);
+                    }
+                    else if (rightNumber < 0)
+                    {
+                        return Math.Pow(leftNumber, rightNumber);
+                    }
+                    else
+                    {
+                        // Both values are some form of integers. The BigInteger implementation is more likely to yield
+                        // a result that is useful for us in this case.
+                        return BigInteger.Pow(leftNumber, rightNumber);
+                    }
+
                 case BANG_EQUAL:
                     return !IsEqual(left, right);
                 case EQUAL_EQUAL:
                     return IsEqual(left, right);
-            }
 
-            // Unreachable.
-            return null;
```

Comment:
This was a really bad idea. You should _never_, ever write code like that, presuming that some "unreachable" code will never be reached. Unless the compiler tells you that is the case, you are more than likely wrong. (In my case, adding a new `enum` value broke this assumption.)